eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -588,22 +589,24 @@ public function createContent(APIContentCreateStruct $contentCreateStruct, array
                         $fieldDefinition,
                         $fieldValue
                     );
-                    if (!empty($fieldErrors)) {
-                        $allFieldErrors[$fieldDefinition->id][$languageCode] = $fieldErrors;
-                    }
-                }
-
-                if (!empty($allFieldErrors)) {
-                    continue;
                 }
 
-                $this->relationProcessor->appendFieldRelations(
+                $relationErrors = $this->relationProcessor->appendFieldRelations(
                     $inputRelations,
                     $locationIdToContentIdMapping,
                     $fieldType,
                     $fieldValue,
-                    $fieldDefinition->id
+                    $fieldDefinition
                 );
+
+                if (!empty($fieldErrors) || !empty($relationErrors)) {
+                    $allFieldErrors[$fieldDefinition->id][$languageCode] = array_merge($fieldErrors, $relationErrors);
+                }
+
+                if (!empty($allFieldErrors)) {
+                    continue;
```

Comment:
afaik it's actually just to ommit some calculations not needed since there are errors.
But yes the code is 99% duplicate between create and update so ripe for refactoring, but then someone will come and say changing protected methods are bad since gg might extend the service to add a hack to it that he never got the time to just contribute in a clean way and the refactoring is blocked, and so the story goes. Good night ;)